samza-core/src/main/java/org/apache/samza/config/ClusterManagerConfig.java

```diff
@@ -189,12 +199,23 @@ public String getContainerManagerClass() {
     return get(CLUSTER_MANAGER_FACTORY, CLUSTER_MANAGER_FACTORY_DEFAULT);
   }
 
-  public boolean getJmxEnabled() {
+  public boolean getJmxEnabledOnJobCoordinator() {
     if (containsKey(CLUSTER_MANAGER_JMX_ENABLED)) {
+      log.info("Configuration {} is deprecated. Please use {}", CLUSTER_MANAGER_JMX_ENABLED, JOB_JMX_ENABLED);
       return getBoolean(CLUSTER_MANAGER_JMX_ENABLED);
     } else if (containsKey(AM_JMX_ENABLED)) {
-      log.info("Configuration {} is deprecated. Please use {}", AM_JMX_ENABLED, CLUSTER_MANAGER_JMX_ENABLED);
+      log.info("Configuration {} is deprecated. Please use {}", AM_JMX_ENABLED, JOB_JMX_ENABLED);
       return getBoolean(AM_JMX_ENABLED);
+    } else if (containsKey(JOB_JMX_ENABLED)) {
+      return getBoolean(JOB_JMX_ENABLED);
+    } else {
+      return true;
+    }
+  }
+
+  public boolean getJmxEnabled() {
```

Comment:
`job.jmx.enabled` should be in JobConfig

Reply:
sure